contracts/value/mod_test.go

```diff
@@ -0,0 +1,235 @@
+	require.NoError(t, err)
+}
+
+func TestCommand_Set(t *testing.T) {
+	contract := NewContract([]byte{}, fakeAccess{})
+
+	cmd := valueCommand{
+		Contract: &contract,
+	}
+
+	err := cmd.set(fake.NewSnapshot(), makeStep(t))
+	require.EqualError(t, err, "'value:key' not found in tx arg")
+
+	err = cmd.set(fake.NewSnapshot(), makeStep(t, KeyArg, "dummy"))
+	require.EqualError(t, err, "'value:value' not found in tx arg")
+
+	err = cmd.set(fake.NewBadSnapshot(), makeStep(t, KeyArg, "dummy", ValueArg, "value"))
+	require.EqualError(t, err, fake.Err("failed to set value"))
+
+	snap := fake.NewSnapshot()
+
+	_, found := contract.index["dummy"]
+	require.False(t, found)
+
+	err = cmd.set(snap, makeStep(t, KeyArg, "dummy", ValueArg, "value"))
+	require.NoError(t, err)
+
+	_, found = contract.index["dummy"]
+	require.True(t, found)
+
+	res, err := snap.Get([]byte("dummy"))
+	require.NoError(t, err)
+	require.Equal(t, "value", string(res))
+}
+
+func TestCommand_Read(t *testing.T) {
+	contract := NewContract([]byte{}, fakeAccess{})
+
+	cmd := valueCommand{
+		Contract: &contract,
+	}
+
+	err := cmd.read(fake.NewSnapshot(), makeStep(t))
+	require.EqualError(t, err, "'value:key' not found in tx arg")
+
+	err = cmd.read(fake.NewBadSnapshot(), makeStep(t, KeyArg, "dummy"))
+	require.EqualError(t, err, fake.Err("failed to get key 'dummy'"))
+
+	snap := fake.NewSnapshot()
+	snap.Set([]byte("dummy"), []byte("value"))
+
+	buf := &bytes.Buffer{}
+	cmd.Contract.printer = buf
+
+	err = cmd.read(snap, makeStep(t, KeyArg, "dummy"))
+	require.NoError(t, err)
+
+	require.Equal(t, "dummy=value", buf.String())
+}
+
+func TestCommand_Delete(t *testing.T) {
+	contract := NewContract([]byte{}, fakeAccess{})
+
+	cmd := valueCommand{
+		Contract: &contract,
+	}
+
+	err := cmd.delete(fake.NewSnapshot(), makeStep(t))
+	require.EqualError(t, err, "'value:key' not found in tx arg")
+
+	err = cmd.delete(fake.NewBadSnapshot(), makeStep(t, KeyArg, "dummy"))
+	require.EqualError(t, err, fake.Err("failed to delete key 'dummy'"))
+
+	snap := fake.NewSnapshot()
+	snap.Set([]byte("dummy"), []byte("value"))
+	contract.index["dummy"] = struct{}{}
+
+	err = cmd.delete(snap, makeStep(t, KeyArg, "dummy"))
+	require.NoError(t, err)
+
+	res, err := snap.Get([]byte("dummy"))
+	require.Nil(t, err)
+	require.Nil(t, res)
+
+	_, found := contract.index["dummy"]
+	require.False(t, found)
+}
+
+func TestCommand_Display(t *testing.T) {
+	contract := NewContract([]byte{}, fakeAccess{})
+	contract.index["key1"] = struct{}{}
+	contract.index["key2"] = struct{}{}
+
+	buf := &bytes.Buffer{}
+	contract.printer = buf
+
+	cmd := valueCommand{
+		Contract: &contract,
+	}
+
+	snap := fake.NewSnapshot()
+	snap.Set([]byte("key1"), []byte("value1"))
+	snap.Set([]byte("key2"), []byte("value2"))
+
+	err := cmd.display(snap)
+	require.NoError(t, err)
+
+	require.Equal(t, "key1=value1,key2=value2", buf.String())
+
+	err = cmd.display(fake.NewBadSnapshot())
+	// we can't assume an order from the map
```

Comment:
how does this comment relate to the assert?

Reply:
I can't check for the exact error string, because it could either say that key1 is missing, or key2.

Reply:
ahhhhh, sorry, now I get it:
```suggestion
	// check for a generic error, exact key depends on undetermined map order
```